src/Collection.php

```diff
@@ -497,6 +498,35 @@ public function dropIndexes(array $options = [])
         return $operation->execute($server);
     }
 
+    /**
+     * Explains explainable commands.
+     *
+     * @see Explain::__construct() for supported options
+     * @see http://docs.mongodb.org/manual/reference/command/explain/
+     * @param Explainable $explainable  Command on which to run explain
+     * @param array       $options      Additional options
+     * @return array|object
+     * @throws UnsupportedException if explainable or options are not supported by the selected server
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function explain($explainable, array $options = [])
+    {
+        if ( ! isset($options['readPreference'])) {
+            $options['readPreference'] = $this->readPreference;
```

Comment:
In addition to using this for server selection, I think Explain should be modified to accept it and pass it to `executeCommand()`. This may call for a new `createOptions()` method:

```php
/**
 * Create options for executing the command.
 *
 * @see http://php.net/manual/en/mongodb-driver-server.executecommand.php
 * @return array
 */
private function createOptions()
{
    $options = [];

    if (isset($this->options['readPreference'])) {
        $options['readPreference'] = $this->options['readPreference'];
    }

    if (isset($this->options['session'])) {
        $options['session'] = $this->options['session'];
    }

    return $options;
}
```

On that note, Explain should also support a `session` option. I forgot about that in the previous PR.

Reply:
I'd suggest using a separate commit under the PHPLIB-156 issue prefix for adding `readPreference` and `session` to the Explain operation. That can still be included in this PR, although you may want to re-order it before the Collection method commit.